### DISCUSSION TITLE
Player Not Loading - Vite + Laravel

### DISCUSSION THREAD

**Discussion by ElisameAraujo** [vidstack/player#1272] [Q&A]
I try to install vidstack via webcomponent using Vite, but the player not loading e can't be played, only the YouTube iframe loads.

![image](https://github.com/vidstack/player/assets/40779799/02c0a2ca-afe1-463d-91ae-4bb69573a88e)

I follow the instructions in the [installation page](https://www.vidstack.io/docs/player/getting-started/installation/web-components?bundler=vite&provider=youtube&styling=plyr-layout) but no luck.

**vite.config.js**
```
import { defineConfig } from 'vite';
import { vite as vidstack } from 'vidstack/plugins';
import laravel from 'laravel-vite-plugin';

export default defineConfig({
    plugins: [
        laravel({
            input: [
                'resources/css/app.css',
                'resources/sass/app.scss',
                'resources/js/app.js',
            ],
            refresh: true,
        }),
        vidstack({})
    ],
});
```

**app.js**
```
import 'bootstrap';
import 'vidstack/bundle';

import Alpine from 'alpinejs';

window.Alpine = Alpine;

Alpine.start();
```

**assistir-video.blade.php**
```
[...]
<div class="row">
        <div class="col-md-9">           
            <media-player title="Sprite Fight" src="youtube/_cMxraX_5RE">
                <media-provider></media-provider>
                <media-video-layout thumbnails="https://files.vidstack.io/sprite-fight/thumbnails.vtt"></media-video-layout>
            </media-player>
            <div class="row">
[...]
```

What I doing wrong?

**Comment by Bahamut-Chocobo** [comment]
Sounds similar to the problem I had.\
Tried to setup Vidstack in Laravel with the steps in this [Getting Started](https://www.vidstack.io/docs/player/getting-started/installation/web-components?bundler=vite&provider=hls&styling=tailwind-css) (difference to your link: provider is `HLS` and styling layout is `Tailwind CSS`).

As part of the setup I added the import to the `app.js` as well
``` js
import 'vidstack/bundle';
```

The dev tools showed me that all media where loaded (cover, captions, HLS playlist, etc), but there was _no_ _way_ _to_ _play_ _it_.

I got around it by adding more imports
``` js
import 'vidstack/bundle';
import 'vidstack/player/ui';
import 'vidstack/player/layouts';
```
… which seems odd when importing the `vidstack/bundle` as the getting started guides states.

> using tailwind, also needed to
> ```js
> import 'vidstack/icons';
> ```

Maybe you could try it as well.

**Reply by hyunbinseo** [reply]
Importing layout fixed the issue in Svelte 5 and Vite 6. Thank you!

```js
import 'vidstack/bundle';
import 'vidstack/player/layouts';
```
